test/contrib/dropbox_test.py

```diff
@@ -0,0 +1,284 @@
+DROPBOX_TEST_OUTER_DIR_TO_CREATE = DROPBOX_TEST_PATH + '/new_folder'
+
+DROPBOX_TEST_DIR_TO_CREATE = DROPBOX_TEST_OUTER_DIR_TO_CREATE + '/inner_folder'
+DROPBOX_TEST_FILE_TO_UPLOAD_BZIP2 = DROPBOX_TEST_PATH + '/bin.file'
+DROPBOX_TEST_FILE_TO_UPLOAD_TEXT = DROPBOX_TEST_PATH + '/text.txt'
+DROPBOX_TEST_FILE_TO_UPLOAD_BIN = DROPBOX_TEST_PATH + '/file.bin'
+DROPBOX_TEST_FILE_TO_UPLOAD_LARGE = DROPBOX_TEST_PATH + '/file.blob'
+
+DROPBOX_TEST_NON_EXISTING_FILE = DROPBOX_TEST_SIMPLE_DIR + 'ajdlkajfal'
+
+
+@attr('dropbox')
+class TestClientDropbox(unittest.TestCase):
+    def setUp(self):
+        self.luigiconn = DropboxClient(DROPBOX_APP_TOKEN)
+
+        self.dropbox_api = dropbox.dropbox.Dropbox(DROPBOX_APP_TOKEN)
+        self.dropbox_api.files_upload(b'hello', DROPBOX_TEST_SIMPLE_FILE)
+        self.dropbox_api.files_upload(b'hello2', DROPBOX_TEST_FILE_IN_DIR)
+        self.dropbox_api.files_upload(b'hello3', DROPBOX_TEST_FILE_TO_MOVE_ORIG)
+        self.dropbox_api.files_upload(b'hello4', DROPBOX_TEST_FILE_TO_COPY_ORIG)
+        self.dropbox_api.files_upload(b'hello3.1', DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.dropbox_api.files_upload(b'hello3.2', DROPBOX_TEST_FILE_TO_DELETE_2)
+
+    def tearDown(self):
+        self.dropbox_api.files_delete_v2(DROPBOX_TEST_PATH)
+        self.dropbox_api._session.close()
+
+    def test_exists(self):
+        self.assertTrue(self.luigiconn.exists('/'))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_PATH))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_DIR))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_FILE))
+
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_listdir_simple(self):
+        list = self.luigiconn.listdir(DROPBOX_TEST_PATH)
+        self.assertTrue('/' not in list)
+        self.assertTrue(DROPBOX_TEST_PATH in list)
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_listdir_multiple(self):
+        list = self.luigiconn.listdir(DROPBOX_TEST_PATH, limit=2)
+        self.assertTrue('/' not in list)
+        self.assertTrue(DROPBOX_TEST_PATH in list)
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_remove(self):
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+        self.luigiconn.remove(DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+        self.luigiconn.remove(DROPBOX_TEST_DIR_TO_DELETE)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+    def test_remove_nonexisting(self):
+        self.assertFalse(self.luigiconn.remove(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_mkdir_new_dir(self):
```

Comment:
what about testing `mkdir` with a trailing `/`? Looks like both of the tested paths do not include trailing slash

Reply:
As I said above, in Dropbox API, trying to create a folder named '/this/is/my/path/' raises an "dropbox.expceptions.ApiError (malformed_path)", because a dropbox path cannot end with /.

We can hide this fact from the user (removing the trailing slash whenever he uses it) or we can only let him use the paths in the format that the API expects.

Additionally, I have created a test to verify and document this behavior
```
    def test_mkdir_new_dir(self):
        ...
        with self.assertRaises(dropbox.exceptions.ApiError):
            self.luigiconn.mkdir(DROPBOX_TEST_TRAILING_SLASH_DIR_TO_CREATE)
            # Path with a trailing slash are considered by Dropbox API as a 'malformed_path'
```

I have chosen this option, but I am open to discuss it.

Reply:
appreciate the additional test

Reply:
I have updated the behavior of mkdir with trailing slashes. 

A new test can be seen [here](https://github.com/joseignaciorc/luigi/blob/1276a26eea80bbfba4460b0948b90b2d9e0cdf2f/test/contrib/dropbox_test.py#L158).